File: php/src/Builder/SQLBuilder/SqlBuilder.php
<?php

namespace Ibrahim13\PdPhp\Builder\SqlBuilder;

use Ibrahim13\PdPhp\Builder\SqlBuilder\Contract\ISqlBuilder;
use Ibrahim13\PdPhp\Builder\SqlBuilder\SqlQuery;
class SqlBuilder implements ISqlBuilder{

	private $sqlQuery;

	public function __constractor(){

		$this->sqlQuery = new SqlQuery();

	}

	public function from(string $table){

		$this->sqlQuery->table = $table;

		return $this;

	}

	public function orderBy(string $column, string $dir = 'DESC'){

		$this->sqlQuery->orderBy = $column . $dir;

		return $this;

	}

	public function limit(int $offset){

		$this->sqlQuery->limit = $offset;

		return $this;

	}

	public function select(array $columns){

		$this->sqlQuery->fields = $columns;
	}

	public function getQuery():SqlQuery{

		return $this->sqlQuery;

	}

}

File: python/builder/sql_builder.py
from builder.Contract.sql_builder_contract import SqlBuilderContract
from .sql_query import SqlQuery


class SqlBuilder(SqlBuilderContract):

    def __init__(self):
        super().__init__()
        self.sql_query = SqlQuery()
    

    def select(self,columns: list)-> SqlBuilderContract:
       self.sql_query.fields = columns
       return self


    def limit(self,limit: int)-> SqlBuilderContract:
        self.sql_query.limit = limit
        return self
    
    
    def where(self,where: list)-> SqlBuilderContract:
        self.sql_query.conditions = where
        return self


    def order_by(self,order: str,dirct ='asc') -> SqlBuilderContract:
        orderBy = f"{order} {dirct}"
        self.sql_query.orderBy = orderBy 
        return self

    
    def from_table(self,table: str)-> SqlBuilderContract:
        self.sql_query.table = table
        return self
    

    def get_query(self) -> SqlQuery:
        return self.sql_query
